src/combinator.rs

```diff
@@ -624,6 +624,107 @@ parser!{
     }
 }
 
+#[derive(Clone)]
+pub struct CountMinMax<F, P> {
+    parser: P,
+    min: usize,
+    max: usize,
+    _marker: PhantomData<fn() -> F>,
+}
+
+impl<P, F> Parser for CountMinMax<F, P>
+where
+    P: Parser,
+    F: FromIterator<P::Output>,
+{
+    type Input = P::Input;
+    type Output = F;
+
+    #[inline]
+    fn parse_lazy(&mut self, input: P::Input) -> ConsumedResult<F, P::Input> {
+        let mut iter = self.parser.by_ref().iter(input.clone());
+        let mut len = 0usize;
+        let value = iter.by_ref().take(self.max).inspect(|_| { len += 1 }).collect();
+        if len < self.min {
+            let message = Error::Message(format!("expected {} more elements", self.min - len).into());
+            ConsumedErr(ParseError::new(input.position(), message))
```

Comment:
No need to track it in inspect as the position in iter.input should be all that we need I think